### PR TITLE
[T3A3][W10-A4] Le Ba Hieu Giang

### PR DESCRIPTION
**Pull request by joeleba**
Ready for review

Changes:

- Added classes `Block`, `Street`, `Unit`, `PostalCode` as subclasses of `AddressComponent`
- Removed attribute `value` from `Address`, as this should be drawn from its components
- Updated `help` messages & unit tests

### CODE REVIEW

File: src/seedu/addressbook/commands/AddCommand.java [hunk 1]
```diff
@@ -18,8 +18,9 @@
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a person to the address book. "
             + "Contact details can be marked private by prepending 'p' to the prefix.\n"
             + "Parameters: NAME [p]p/PHONE [p]e/EMAIL [p]a/ADDRESS  [t/TAG]...\n"
```
Comment (damithc) [inline src/seedu/addressbook/commands/AddCommand.java]: The address format needs to be updated too, right?

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -8,13 +8,19 @@
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    public static final String EXAMPLE = "123, Clementi Ave 3, #12-34, 231534";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses must be of the following format: BLOCK, STREET, UNIT, POSTAL_CODE";
+    public static final String ADDRESS_VALIDATION_REGEX = "(.+), (.+), (.+), (.+)";
```
Comment (damithc) [inline src/seedu/addressbook/data/person/Address.java]: Good job attempting to use regex. :+1:

File: src/seedu/addressbook/data/person/AddressComponent.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+    protected String value;
+
+    public AddressComponent(String value) {
+        this.value = value;
+    }
+
+    /**
+     * Getter
+     */
```
Comment (damithc) [inline src/seedu/addressbook/data/person/AddressComponent.java]: Comment redundant?

**Comment by damithc** [comment]
Good job @joeleba 
Some comments added. Please close the PR after reading comments.

**Comment by joeleba** [comment]
Thanks for your comments prof!

**Pull request closed without merging**